src/engine.cc

```diff
@@ -0,0 +1,83 @@
+#include "engine.h"
+#include "gobject.h"
+
+#include <eos-knowledge-content.h>
+
+using namespace v8;
+
+struct CallbackData {
+  Persistent<Promise::Resolver> resolver;
+};
+
+static void EngineGetObjectCallback(GObject *source,
+                                    GAsyncResult *result,
+                                    gpointer user_data)
+{
+  Isolate *isolate = Isolate::GetCurrent ();
+  HandleScope scope(isolate);
+  CallbackData *data = (CallbackData *)user_data;
+
+  GError *error = NULL;
+  EkncContentObjectModel *model;
+
+  Local<Promise::Resolver> resolver = Local<Promise::Resolver>::New(isolate, data->resolver);
```

Comment:
I'll try it. My understanding of the v8 api is pretty damn week. I only got as deep as I needed to get things working. I was trying to mirror the structure of the function code this promise code just replaced, which made a scope, then a local out of the persistent and called into that. But maybe that's not necessary at all?